### ISSUE TITLE
arm64 and xnu blockers?

### ISSUE THREAD

**Issue by chadbrewbaker** [aws/aws-lambda-runtime-interface-emulator#5]
"The component supports only Linux x84-64 architectures"

What is the blocker to running on macos_arm64? Quick glance I didn't see any x86 specific dependencies or Linux specific threading.  Is there something I missed?

**Comment by jfuss** [comment]
The Runtime Interface Emulator is only built for the OS Lambda Supports. Since this is meant to run within the container and lambda only supports x86, there is no need for arm, xnu, etc.

Closing

**Comment by chadbrewbaker** [comment]
The purpose of an emulator is to abstract. I just asked if there were any blockers you knew of.

If you are going to weigh it down with linux system calls and x86 then you might as well open source one of the full stacks modulo crypto so Daniel Lemire etc can SIMD the HTML/JSON transforms.

**Comment by jfuss** [comment]
@chadbrewbaker I am not really following the second part there. The blocker is that Lambda doesn't support anything but x86. The goal of this project is to allow you to test you functions locally. If we allow anything but what Lambda's OS currently supports, then you could run something locally that does not work on Lambda itself. The restrictions do not have anything to do with syscalls but more to do with the fidelity with Lambda.

**Comment by klische** [comment]
In

> @chadbrewbaker I am not really following the second part there. The blocker is that Lambda doesn't support anything but x86. The goal of this project is to allow you to test you functions locally. If we allow anything but what Lambda's OS currently supports, then you could run something locally that does not work on Lambda itself. The restrictions do not have anything to do with syscalls but more to do with the fidelity with Lambda.

This logic does not make sense. In order to test locally on a machine that is ARM64 (M1 Macs), the RIE images must run in a docker for ARM64. Otherwise, the entirety of the AWS developer documentation “on debbugging Lambdas locally” is broken for ARM64/M1 Macs.

**Comment by rajasimon** [comment]
Wish this AWS Lambda docker works natively in my M1 machine. Now that it's not curios about whether SAM supports or not?

**Comment by jfuss** [comment]
@klische That is not true. When running on an M1, you still need to run the x86 container. RIE is not meant to run on anything but what Lambda Supports. Lambda does not support ARM therefore producing an ARM artifact doesn't give you anything. Even if we did provide an ARM artifact, that would allow you to run in ARM which Lambda doesn't support. So testing on ARM but running on x86 can lead to incompatibilities and why we decided to only produce artifacts for Lambda Supported OS's.

@rajasimon SAM CLI supports M1 and runs with RIE. So it should be working. What SAM CLI does, is adds RIE into the container (through a docker build process) and then mounts the code into the container for invoking.
